### PR TITLE
Fix applying code actions that change AdditionalDocuments

### PR DESCRIPTION
**Pull request by jasonmalinowski**
Some code was accidentally calling GetDocument with an additional document ID; this didn't end well.

### CODE REVIEW

**Comment by CyrusNajmabadi** [comment]
#innerjason: Any way we can have a test for this?

**Comment by jasonmalinowski** [comment]
@CyrusNajmabadi I wholeheartedly agree....any idea where the tests are? Or maybe @heejaechang or @dpoeschl know?

File: src/Workspaces/Core/Portable/Workspace/Solution/Document.cs [hunk 1]
```diff
@@ -41,10 +41,13 @@ public partial class Document : TextDocument
         /// <summary>
         /// True if the info of the document change (name, folders, file path; not the content)
         /// </summary>
-        internal bool HasInfoChanged(Document otherDocument)
+        internal override bool HasInfoChanged(TextDocument otherTextDocument)
         {
-            return DocumentState.Attributes != otherDocument.DocumentState.Attributes
-                || DocumentState.SourceCodeKind != otherDocument.SourceCodeKind;
+            var otherDocument = otherTextDocument as Document ??
+                throw new ArgumentException("otherTextDocument isn't a regular document.", nameof(otherTextDocument));
```
Comment (sharwell) [inline src/Workspaces/Core/Portable/Workspace/Solution/Document.cs]: :bulb: string interpolation with nameof

**Comment by jasonmalinowski** [comment]
And oh I forgot to say yesterday: I'm going to add an integration test for this, since we've also made mistakes on the VS side of things so we can cover the end-to-end scenario. I wanted the fix in before that (which may take a few days to write) because this is also blocking dogfooding -- it means the public API analyzer fixes are useless, and it's painful working without those.

**Comment by CyrusNajmabadi** [comment]
Is tehre a workaround in the meantime?  Thanks!

**Comment by jasonmalinowski** [comment]
I guess I'm not sure if this fix made it into Preview 3 or Preview 4. If it's not in Preview 3 then the workaround @sharwell described in #35670 was actually the same workaround I was doing as well. :smile:

**Pull request merged**